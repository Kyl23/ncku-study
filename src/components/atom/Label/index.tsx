import { NextPage } from 'next';
import ClickEventListener from '../ClickEventListener';
import PropsType from '../propsType';
import { DeleteButton, Title } from './style';

const Label: NextPage<PropsType> = ({ styleProps, textProps, eventProps }) => {
  const { value } = textProps;
  const { enableDelete, onDelete } = eventProps;
  return (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <Title {...styleProps}>
      {value}{' '}
      {enableDelete && (
        <ClickEventListener onClick={onDelete}>
          <DeleteButton />
        </ClickEventListener>
      )}
    </Title>
  );
};

export default Label;
