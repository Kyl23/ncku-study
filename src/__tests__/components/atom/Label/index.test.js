import Label from '@/components/atom/Label';
import { fireEvent, render, screen } from '@testing-library/react';

describe('test for atom - Label', () => {
    it('render Label without props must be same', () => {
        const { container } = render(<Label />);

        expect(container).toMatchSnapshot();
    });

    it('render Label with specify textProps', () => {
        const value = 'this is a test value';
        const { container } = render(<Label textProps={{ value }} />);

        expect(screen.getByText(value)).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    it('render Label with specify eventProps (enable delete button)', () => {
        const enableDelete = true;
        const mockDelete = jest.fn();
        const { container } = render(
            <Label eventProps={{ enableDelete, onDelete: mockDelete }} />
        );
        mockDelete.mockClear();
        expect(mockDelete).toBeCalledTimes(0);
        fireEvent.click(screen.getByRole('button'));

        expect(mockDelete).toBeCalledTimes(1);
        expect(container).toMatchSnapshot();
    });

    it('render Label with specify eventProps (disable delete button)', () => {
        const enableDelete = false;
        const { container } = render(<Label eventProps={{ enableDelete }} />);

        expect(screen.queryByRole('button')).toBeFalsy();
        expect(container).toMatchSnapshot();
    });
});
