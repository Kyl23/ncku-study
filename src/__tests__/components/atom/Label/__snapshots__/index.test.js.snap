// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for atom - Label render Label with specify eventProps (disable delete button) 1`] = `
<div>
  <div
    class="css-1e07x57"
  >
     
  </div>
</div>
`;

exports[`test for atom - Label render Label with specify eventProps (enable delete button) 1`] = `
<div>
  <div
    class="css-1e07x57"
  >
     
    <button
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15wd22y-MuiSvgIcon-root"
        data-testid="HighlightOffIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.59 8 12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`test for atom - Label render Label with specify textProps 1`] = `
<div>
  <div
    class="css-1e07x57"
  >
    this is a test value
     
  </div>
</div>
`;

exports[`test for atom - Label render Label without props must be same 1`] = `
<div>
  <div
    class="css-1e07x57"
  >
     
  </div>
</div>
`;
